xarray/backends/api.py

```diff
@@ -474,6 +475,15 @@ def open_dataset(
             "all other options must be passed as keyword arguments"
         )
 
+    # TODO remove after v0.19
+    if kwargs.pop("lock", None):
+        warnings.warn(
+            "The kwarg 'lock' has been deprecated, and is now"
+            "ignored. In future (from v0.19) passing lock will "
```

Comment:
Unless we think we're conscientious enough to hit these, maybe better to leave out?

```suggestion
            "ignored. In the future passing lock will "
```